common/utilities/tests/test_timing.py

```diff
@@ -0,0 +1,159 @@
+import asyncio
+import json
+import logging
+import math
+import time
+from unittest import TestCase
+from unittest.mock import patch, Mock
+
+from tornado.testing import AsyncHTTPTestCase
+from tornado.web import Application, RequestHandler
+
+from common.utilities import timing
+from common.utilities.test_utilities import async_test
+
+
+class TestTimeUtilities(TestCase):
```

Comment:
In terms of coverage, you've got tests for lots of different scenarios for `time_method`, but none for the same scenarios on the other methods, and they all implement the logic to handle these scenarios separately. If these methods are refactored to re-use some more code then this should take care of most of this coverage, but worth bearing in mind if certain bits of logic remain within the individual methods.

Reply:
So I've reorganised a bit now, most tests now include 2 sub tests, one sync and one async version, this should improve the general coverage and make the tests a bit cleaner as well.